### DISCUSSION TITLE
NIMBLE BLE EVENT DESCRIPTOR  battery_descriptor callback function does not work (ESP_IDF_4.4.1)

### DISCUSSION THREAD

**Discussion by KumphanartDansiri** [Mair/esp32-course#98] [Course Qeustions]
I follow the course on BLE Event and Descriptor.  Everything compile OK.  I use BLE Scanner on Android.  
- The Battery Service shows up with Read, Notify correctly.
- 2 Client Characteristics Configurations show up with Read Permission Only.  The first client can read Notify Status.
- There is no Notify OFF Button like in your video.
- When Tab notify in battery service, Battery Percentage decrease correctly.  And the first client characteristics show Notify Enable.
- When Tab notify in battery service.  Battery Percentage on the phone stop decreasing.  The first client characteristics show Notify Disabled.  But Computer Monitor Console, Battery Percentage decreasing.  Every time Notify Icon is pressed, Monitor console will show EVENT_SUBSCRIBE.
- I put ESP_LOGI in battery_descriptor call back function.  It seems, the function has never been called.

How to diagnose further?

**Comment by Mair** [comment]
Not sure. Have you tried to run the code in the repo?

**Reply by KumphanartDansiri** [reply]
Code or Repo Shows the same result.  Battery Level does not stop count down.

**Reply by Alexx-Z** [reply]
I also encountered the same problem. I think there is a problem in the code (maybe new changes occurred in the BLE protocol).

We could not find Client Characteristic Configuration (GATT Descriptor) on the Bluetooth website, and the link that was provided was also not working.

The link provided on the code: https://www.bluetooth.com/wp-content/uploads/Sitecore-Media-Library/Gatt/Xml/Descriptors/org.bluetooth.descriptor.gatt.client_characteristic_configuration.xml

Could you provide a new working code for BLE-Notify with comments?

**Comment by Mair** [comment]
yeah, that's changed a lot since I did the video.
Try this out https://www.bluetooth.com/specifications/specs/
each one of those links goes to a document where you can get the info required.
